### PR TITLE
util: introduce Event

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @siddontang PTAL

### CODE REVIEW

File: src/raftstore/store/msg.rs [hunk 1]
```diff
@@ -88,40 +89,28 @@ impl fmt::Debug for Msg {
...
 // We should know that even timeout happens, the command may still
 // be handled in store later.
 pub fn call_command(sendch: &SendCh,
                     request: RaftCmdRequest,
                     timeout: Duration)
                     -> Result<RaftCmdResponse> {
-    let resp: Option<RaftCmdResponse> = None;
-    let pair = Arc::new((Mutex::new(resp), Condvar::new()));
-    let pair2 = pair.clone();
+    let finished = Arc::new(Event::new());
```
Comment (ngaut) [inline src/raftstore/store/msg.rs]: Can we simply use ev instead of finished ?

Comment (BusyJay) [inline src/raftstore/store/msg.rs]: `finished` here means finish event.

File: src/storage/engine/raftkv.rs [hunk 2]
```diff
@@ -104,32 +105,21 @@ impl<T: PdClient, Trans: Transport> RaftKv<T, Trans> {
...
-        let &(ref lock, ref cvar) = &*pair;
-        let mut v = lock.lock().unwrap();
-        while v.is_none() {
-            let (resp, timeout_res) = cvar.wait_timeout(v, timeout).unwrap();
-            if timeout_res.timed_out() {
-                return Err(Error::Timeout(timeout));
-            }
-
-            v = resp
+        if finished.wait_set(Some(timeout)) {
```
Comment (ngaut) [inline src/storage/engine/raftkv.rs]: s/wait_set/wait_timeout/

File: src/util/event.rs [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            }
+            l = v;
+        }
+        true
+    }
+
+    /// Wait for set status to be clear.
+    ///
+    /// If it's clear before timeout, true is returned; otherwise return false.
+    pub fn wait_take(&self, timeout: Option<Duration>) -> bool {
```
Comment (ngaut) [inline src/util/event.rs]: s/wait_take/wait_clear/

**Comment by ngaut** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**